macos/Sources/App/macOS/AppDelegate.swift

```diff
@@ -509,8 +520,15 @@ class AppDelegate: NSObject,
     }
 
     @objc private func ghosttyBellDidRing(_ notification: Notification) {
-        // Bounce the dock icon if we're not focused.
-        NSApp.requestUserAttention(.informationalRequest)
+        // Only request attention and set badge when the app is not active
+        if !NSApp.isActive {
+            // Bounce the dock icon if we're not focused.
+            NSApp.requestUserAttention(.informationalRequest)
```

Comment:
We don't need to guard this one because it automatically does nothing if the app is active. We should just let macOS do it.